src/main/java/de/srendi/advancedperipherals/common/util/TeleportUtil.java

```diff
@@ -0,0 +1,41 @@
+package de.srendi.advancedperipherals.common.util;
+
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.phys.Vec3;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class TeleportUtil {
+    private TeleportUtil() {}
+
+    public static <T extends Entity> T teleportToWithPassengers(T entity, ServerLevel newLevel, Vec3 newPos) {
+        Vec3 oldPos = entity.position();
+        List<Entity> passengers = new ArrayList<>(entity.getPassengers());
+        T newEntity;
+        if (entity instanceof ServerPlayer player) {
+            // TODO <1.20.1>: player will be reconstruct in 1.20.1
```

Comment:
What exactly do you mean with this todo? Is this a breaking change for 1.20.1?

Reply:
like in 1.20.1 ServerPlayer will reconstruct itself instead of using same instance in teleporting